src/schema/field_type.rs

```diff
@@ -63,31 +62,28 @@ impl FieldType {
                     }
                 }
             }
-            Json::U64(ref field_val_u64) => {
+            JsonValue::Number(ref field_val_num) => {
                 match *self {
                     FieldType::I64(_) => {
-                        if *field_val_u64 > (i64::max_value() as u64) {
-                            Err(ValueParsingError::OverflowError(format!("Value {:?} is too high for a i64.", field_val_u64)))
+                        if let Some(field_val_u64) = field_val_num.as_u64() {
```

Comment:
This can use [`field_val_num.as_i64()`](https://docs.serde.rs/serde_json/struct.Number.html#method.as_i64) which will do the bounds check and return `None` if it's not an `i64`.